lib/ui/widgets/food_alergens_text_field.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:flutter/cupertino.dart';
+import 'package:zachranobed/extensions/build_context_extensions.dart';
+import 'package:zachranobed/ui/widgets/text_field.dart';
+
+import '../../enums/food_form_field_type.dart';
+import 'form/form_validation_manager.dart';
+
+class FoodAllergensTextField extends StatefulWidget {
+  final int index;
+  final String label;
+  final String? initialValue;
+  final Function(String) onChanged;
+  final FormValidationManager formValidationManager;
+
+  const FoodAllergensTextField({
+    super.key,
+    required this.index,
+    required this.label,
+    required this.onChanged,
+    required this.formValidationManager,
+    this.initialValue,
+  });
+
+  @override
+  _FoodAllergensTextFieldState createState() => _FoodAllergensTextFieldState();
```

Comment:
IDE shows a hint here, that it should be `State<FoodAllergensTextField>` for return type 👀